### PR TITLE
Avoid logging exception for bad table configuration

### PR DESCRIPTION
**Pull request by antumbde**
For bad table configuration where number of replicas is greater than
number of instances or for invalid replicasPerPartition, log info and
return 400. Do not log exception.

### CODE REVIEW

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableIndexingConfigs.java [hunk 1]
```diff
@@ -55,7 +54,12 @@ public Representation put(Representation entity) {
     }
     try {
       return updateIndexingConfig(tableName, entity);
-    } catch (final Exception e) {
+    } catch (PinotHelixResourceManager.InvalidTableConfigException e) {
+      LOGGER.info("Failed to update metadata configuration for table {}, error: {}", tableName, e.getMessage());
+      ControllerRestApplication.getControllerMetrics().addMeteredGlobalValue(
+          ControllerMeter.CONTROLLER_TABLE_UPDATE_ERROR, 1L);
+      return errorResponseRepresentation(Status.CLIENT_ERROR_BAD_REQUEST, e.getMessage());
+    }catch (final Exception e) {
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableIndexingConfigs.java]: nit: add extra space before 'catch'

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableIndexingConfigs.java [hunk 2]
```diff
@@ -55,7 +54,12 @@ public Representation put(Representation entity) {
...
       LOGGER.error("Caught exception while updating indexing configs for table {}", tableName, e);
       ControllerRestApplication.getControllerMetrics().addMeteredGlobalValue(ControllerMeter.CONTROLLER_TABLE_INDEXING_GET_ERROR, 1L);
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableIndexingConfigs.java]: don't we want to bump CONTROLLER_TABLE_UPDATE_ERROR here as well?

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableMetadataConfigs.java [hunk 3]
```diff
@@ -46,6 +48,11 @@ public Representation put(Representation entity) {
 
     try {
       return updateTableMetadata(tableName, entity);
+    } catch (PinotHelixResourceManager.InvalidTableConfigException e) {
+      LOGGER.info("Failed to update metadata configuration for table {}, error: {}", tableName, e.getMessage());
+      ControllerRestApplication.getControllerMetrics().addMeteredGlobalValue(
+          ControllerMeter.CONTROLLER_TABLE_UPDATE_ERROR, 1L);
+      return errorResponseRepresentation(Status.CLIENT_ERROR_BAD_REQUEST, e.getMessage());
     } catch (final Exception e) {
       LOGGER.error("Caught exception while updating metadata for table {}", tableName, e);
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableMetadataConfigs.java]: increment CONTROLLER_TABLE_UPDATE_ERROR here?

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResource.java [hunk 4]
```diff
@@ -391,10 +391,18 @@ public Representation updateTableConfig(@Parameter(name = "tableName", in="path"
...
+      ControllerRestApplication.getControllerMetrics().addMeteredGlobalValue(
+          ControllerMeter.CONTROLLER_TABLE_UPDATE_ERROR, 1L);
+      return errorResponseRepresentation(Status.CLIENT_ERROR_BAD_REQUEST, e.getMessage());
     } catch (IOException e) {
       LOGGER.error("Failed to update table configuration for table: {}", tableName, e);
+      ControllerRestApplication.getControllerMetrics().addMeteredGlobalValue(
+          ControllerMeter.CONTROLLER_TABLE_UPDATE_ERROR, 1L);
       return errorResponseRepresentation(Status.SERVER_ERROR_INTERNAL,
           "Internal error while updating table configuration");
+
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResource.java]: Do we want to add a catch all exception here?

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableSegmentConfigs.java [hunk 5]
```diff
@@ -51,6 +50,11 @@ public Representation put(Representation entity) {
...
     try {
       return updateSegmentConfig(tableName, entity);
+    } catch (PinotHelixResourceManager.InvalidTableConfigException e) {
+      LOGGER.info("Failed to update metadata configuration for table {}, error: {}", tableName, e.getMessage());
+      ControllerRestApplication.getControllerMetrics().addMeteredGlobalValue(
+          ControllerMeter.CONTROLLER_TABLE_UPDATE_ERROR, 1L);
+      return errorResponseRepresentation(Status.CLIENT_ERROR_BAD_REQUEST, e.getMessage());
     } catch (final Exception e) {
       LOGGER.error("Caught exception while updating segment config ", e);
       ControllerRestApplication.getControllerMetrics().addMeteredGlobalValue(ControllerMeter.CONTROLLER_TABLE_SCHEMA_UPDATE_ERROR, 1L);
```
Comment (mcvsubbu) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableSegmentConfigs.java]: increment CONTROLLER_TABLE_UPDATE_ERROR?

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableIndexingConfigs.java (hunk 1 above)
Comment (antumbde) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableIndexingConfigs.java]: done

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableSegmentConfigs.java (hunk 5 above)
Comment (antumbde) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableSegmentConfigs.java]: Existing metric is wrong here. Replaced with table_update_error

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableMetadataConfigs.java (hunk 3 above)
Comment (antumbde) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableMetadataConfigs.java]: done

File: pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResource.java (hunk 4 above)
Comment (antumbde) [inline pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotTableRestletResource.java]: done

**Pull request merged**